pkgs/by-name/al/alioth/package.nix

```diff
@@ -0,0 +1,32 @@
+{ lib, fetchFromGitHub, rustPlatform }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "alioth";
+  version = "0.5.0";
+
+  src = fetchFromGitHub {
+    owner = "google";
+    repo = "alioth";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-K0Hx6EJYyPJZA+FLIj44BtUuZZOqWW2DUJt1QbeZyu0=";
+  };
+
+  # Checks run with the release build
+  postPatch = ''
+    substituteInPlace alioth/src/utils/utils.rs \
+      --replace-fail "debug_assert_eq!" "assert_eq!"
```

Comment:
Not sure this is the right fix — upstream intentionally doesn't want those asserts running at runtime.

Maybe we should set `cargoCheckType = "debug"`, with a comment?

Reply:
@lencerf Please advise

Reply:
`debug_assert` was used because I think those assertion violations should be addressed at coding time and can be caught in tests (but I have to admit that current alioth test courage is very low). In `utils.rs`, `debug_assert_eq` is used in `align_up` and `align_down` to check the parameter `align`, and in most cases `align_up`/`align_down` take hard coded values. The only exception is in `search_pvh_note` of `xen.rs` where `align` comes from the input, I will fix that later (https://github.com/google/alioth/issues/174). 
In the future `debug_assert_eq` may appear in some performance critical paths (while currently they are not), replacing `debug_assert_eq` with `assert_eq` will affect runtime performance.

So in general, I do not think you should change `debug_assert_eq` to `assert_eq`. 

I am familiar with `checkType = "debug"`, is it some nix config?

Reply:
> I am familiar with `checkType = "debug"`, is it some nix config?

Yes.  By default, in Nixpkgs cargo tests are run in release mode, to avoid having to build everything once in debug mode for tests, and once in release mode for installation.  Setting `checkType = "debug"` builds the tests in debug mode instead, which sounds like it's the right thing to do here.

Reply:
Thanks to both of you! I went with `checkType`.